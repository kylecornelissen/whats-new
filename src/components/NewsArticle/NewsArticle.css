img {
  width: 300px;
}

a {
  background: magenta;
  color: white;
  text-decoration: none;
}
