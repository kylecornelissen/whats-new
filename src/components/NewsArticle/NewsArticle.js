import React from 'react';
import './NewsArticle.css';

const NewsArticle = (props) => {
  return (
    <article>
      <img src={props.article.img} alt="news article" />
      <h2>{props.article.headline}</h2>
      <p>{props.article.description}</p>
      <div className="article-link-btn"><a className="article-link" href={props.article.url}>Link to Article â†’</a></div>
    </article>
  )
};

export default NewsArticle;
