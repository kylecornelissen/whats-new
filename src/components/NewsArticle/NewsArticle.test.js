import React from 'react';
import { shallow } from 'enzyme';
import local from '../../data/local';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
  it('should match a snapshot of the NewsArticle', () => {
    const wrapper = shallow(<NewsArticle article={local[0]} key={42}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
