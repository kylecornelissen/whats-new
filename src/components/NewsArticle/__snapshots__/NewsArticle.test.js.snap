// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match a snapshot of the NewsArticle 1`] = `
<article>
  <img
    alt="news article"
    src="https://theknow.denverpost.com/wp-content/uploads/2016/03/CDXXTHEWHODSC_4082x.jpg"
  />
  <h2>
    The Who postpones Denver Concert at the Pepsi Center
  </h2>
  <p>
    Classic-rock band The Who has postponed its Denver concert, set for Sunday, Sept. 29, after an illness robbed singer Roger Daltrey of his voice, the band said on Twitter.
  </p>
  <div
    className="article-link-btn"
  >
    <a
      className="article-link"
      href="https://theknow.denverpost.com/2019/09/27/the-who-postpones-denver-concert/225182/"
    >
      Link to Article â†’
    </a>
  </div>
</article>
`;
